core/src/com/unciv/logic/city/CityStats.kt

```diff
@@ -122,13 +124,44 @@ class CityStats {
         val stats = Stats()
 
         for (otherCiv in cityInfo.civInfo.getKnownCivs()) {
-            if (otherCiv.isCityState() && otherCiv.cityStateType == CityStateType.Maritime
-                    && otherCiv.getDiplomacyManager(cityInfo.civInfo).relationshipLevel() >= RelationshipLevel.Friend) {
-                if (cityInfo.isCapital()) stats.food += 3
-                else stats.food += 1
+            if (otherCiv.isCityState() && otherCiv.getDiplomacyManager(cityInfo.civInfo).relationshipLevel() >= RelationshipLevel.Friend) {
+                val eraInfo = cityInfo.civInfo.getEraObject()
+
+                if (eraInfo.friendBonus[otherCiv.cityStateType.name] == null || eraInfo.allyBonus[otherCiv.cityStateType.name] == null) {
+                    // Deprecated, assume Civ V values for compatibility
+                    if (otherCiv.cityStateType == CityStateType.Maritime && otherCiv.getDiplomacyManager(cityInfo.civInfo).relationshipLevel() == RelationshipLevel.Ally)
+                        stats.food += 1
+                    if (otherCiv.cityStateType == CityStateType.Maritime && cityInfo.isCapital())
+                        stats.food += 2
+
+                    if (cityInfo.civInfo.hasUnique("Food and Culture from Friendly City-States are increased by 50%"))
+                        stats.food *= 1.5f
+                    return stats
+                }
+
+                for (bonus in if (otherCiv.getDiplomacyManager(cityInfo.civInfo).relationshipLevel() == RelationshipLevel.Friend)
+                    eraInfo.friendBonus[otherCiv.cityStateType.name]!! else eraInfo.allyBonus[otherCiv.cityStateType.name]!!) {
+                    if (bonus.getPlaceholderText() == "Provides [] [] []" && cityInfo.matchesFilter(bonus.getPlaceholderParameters()[2])) {
+                        val stattoadd = when (bonus.getPlaceholderParameters()[1]) {
+                            "Food" -> Stat.Food
+                            "Production" -> Stat.Production
+                            "Gold" -> Stat.Gold
+                            "Faith" -> Stat.Faith
+                            "Science" -> Stat.Science
+                            "Culture" -> Stat.Culture
+                            "Happiness" -> Stat.Happiness
+                            else -> null
+                        }
```

Comment:
Stat.valueOf(), and assume that it's fine. Better to crash if he has a misconfigured unique than to stay silent.

Reply:
Um... `Stat.values().firstOrNull { it.name == whatever } ?: default` isn't really worse performance-wise than `try { Stat.valueOf(whatever) } catch (ex: Exception) { default }`, if you ever wonder.

... but shouldn't alarm bells ring when something involving Unique and Stat does not go through `Stats.parse()`?

Would make code shorter, unique more flexible, and get rid of 1 green warning - the spellcheck on the variable name :rofl: